definitions/scenarios/self_upgrade.rb

```diff
@@ -65,6 +65,23 @@ class SelfUpgrade < SelfUpgradeBase
     end
 
     def downstream_self_upgrade(pkgs_to_update)
+      unless self_upgrade_allowed?
+        if version_difference > 1
```

Comment:
Do we need the complicated difference computation, or would the following suffice?
```suggestion
        if Gem::Version.new(current_version).bump != Gem::Version.new(target_version)
```